cmake/reports/CMakeLists.txt

```diff
@@ -3,7 +3,11 @@
 set(flag_for_ram_report ram)
 set(flag_for_rom_report rom)
 set(flag_for_footprint all -q)
-set(report_depth 99)
+
+set(report_depth 99 CACHE STRING "Depth of tree to display")
+set(sort_key "name" CACHE STRING "Sort key for results display (name or size)")
+set_property(CACHE sort_key PROPERTY STRINGS "name" "size")
```

Comment:
So a full rebuild is required to change these? I cannot accept that limitation.

Reply:
Firstly, thank you both for the review.

I must admit I am somewhat of a novice at using CMake, however from my testing of the script it seemed like this was the appropriate way to pass the parameters through to the script. As @tejlmand said, it is a much shorter build that a full rebuild.

I hadn't considered Kconfig, but it seems like this would be an option that you might want to change without the full recompile. However, I'd be happy to defer to both of your expertise on the best method to implement this.

Reply:
I would normally prefer Kconfig, as that is mostly used, but exactly for the concern raised by @nordicjm, and which I share, then I think a CMake variable is the best choice in this case.

would like to wait for @nordicjm to see if his concerns has been addressed.